pandas/tests/series/methods/test_fillna.py

```diff
@@ -671,6 +671,27 @@ def test_fillna_categorical_with_new_categories(self, fill_value, expected_outpu
         result = ser.fillna(fill_value)
         tm.assert_series_equal(result, exp)
 
+    @pytest.mark.parametrize(
+        "fill_value, expected_output",
+        [
+            ("B", ["A", "B", "B", "B", "C"]),
+            ("C", ["A", "B", "C", "C", "C"])
+        ],
+    )
+
+    def test_series_fill(fill_value, expected_output):
+        # GH32414
+        data = ["A", "B", np.nan, np.nan, "C"]
+        cat = Categorical(data, categories=["A", "B", "C"])
+        ser = Series(cat)
+        exp = Categorical(expected_output, categories=["A", "B", "C"])
+        exp_ser = Series(exp)
+        result = ser.fillna(fill_value)
+        filled = cat.fillna(fill_value)
+        tm.assert_almost_equal(result, exp_ser)
```

Comment:
can you use
`tm.assert_series_equal(result_ser, exp_ser)`

and
`tm.assert_categorical_equal(result_cat, exp_cat)`

Reply:
and rename things a bit, this is very hard to read

Reply:
yaa i surely rename it. Thanks for checking that out.